### PR TITLE
use local siad

### PR DESCRIPTION
**Pull request by lukechampine**
Sia-UI now calls (by default) `./Sia/siad` instead of just `siad`. This lets us bundle siad with Sia-UI instead of requiring it to be in the user's $PATH. I'm not sure how we should go about adding the actual binary to the project though. A submodule/subtree doesn't really make sense, we just want the binary.

### CODE REVIEW

**Comment by Mingling94** [comment]
Bundling siad with Sia-UI would simplify a lot of siad handling such as with #76 though I wasn't sure to begin with why we have Sia-UI dependent on these binaries outside of it. Getting it to work with Windows won't be too difficult since we can work with relative addresses instead of checking $PATH.

**Comment by cyrozap** [comment]
Maybe we could have Sia-UI try to start a bundled `siad` and then, failing that, try to run `siad` in the user's $PATH. This way, users could still install the UI via `npm` (where no `siad` will be bundled) or they could download the all-inclusive packaged app and we wouldn't have to worry about altering any code before packaging it.

**Comment by lukechampine** [comment]
Yeah, I was thinking along the same lines but I wasn't exactly sure how to implement it.

**Comment by lukechampine** [comment]
hmm. I ran into another problem. Using `./Sia/siad` only works if you're running Sia-UI from the right folder. If you launch by, e.g., double-clicking it in a file browser, the relative path will be wrong.

I was thinking you could fix this by putting the absolute path in config.json. But how would you figure out the correct absolute path in the first place?

**Comment by cyrozap** [comment]
Maybe the Node [path](https://nodejs.org/api/all.html#all_path) module would work for that? Something similar to how you can get the directory a bash script is running in from within the script itself.

**Comment by lukechampine** [comment]
I actually wound up using `__dirname`, seems to do the job well enough.

File: main.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ function loadConfig() {
     } else {
         var defaultConfigJSON = {
             "autogenerated": true,
-            "siad_cmd": "siad",
+            "siad_cmd": "./siad",
```
Comment (cyrozap) [inline main.js]: This line still has the relative path.

Comment (lukechampine) [inline main.js]: true, but it's a non-issue. [What actually happens](https://github.com/NebulousLabs/Sia-UI/pull/89/files#diff-5c2754430010d5f3e95b45e1bb36d58bR35) is that the command `./siad` is executed in the `Sia/` directory. It's equivalent to `cd [absolute path] && ./siad`.

Comment (cyrozap) [inline main.js]: Ah, I see--I was just concerned it might break Windows compatibility.

**Comment by Mingling94** [comment]
So #14 is an issue again with this fix, at least on Windows.

**Pull request merged**